packages/contracts/contracts/L1/ValidatorManager.sol

```diff
@@ -0,0 +1,497 @@
+     * @notice The address of the trusted validator.
+     */
+    address public immutable TRUSTED_VALIDATOR;
+
+    /**
+     * @notice The minimum duration to change the commission rate of the validator (in seconds).
+     */
+    uint128 public immutable COMMISSION_RATE_MIN_CHANGE_SECONDS;
+
+    /**
+     * @notice The duration of a submission round for one output (in seconds).
+     *         Note that there are two submission rounds for an output: PRIORITY ROUND and PUBLIC
+     *         ROUND.
+     */
+    uint128 public immutable ROUND_DURATION_SECONDS;
+
+    /**
+     * @notice The minimum duration to get out of jail (in seconds).
+     */
+    uint128 public immutable JAIL_PERIOD_SECONDS;
+
+    /**
+     * @notice Maximum allowed number of output non-submissions in priority round before the
+     *         validator goes to jail.
+     */
+    uint128 public immutable JAIL_THRESHOLD;
+
+    /**
+     * @notice Address of the next validator with priority for submitting output.
+     */
+    address internal _nextPriorityValidator;
+
+    /**
+     * @notice A mapping of the jailed validator to the jail expiration timestamp.
+     */
+    mapping(address => uint128) internal _jail;
+
+    /**
+     * @notice A modifier that only allows L2OutputOracle contract to call.
+     */
+    modifier onlyL2OutputOracle() {
+        require(
+            msg.sender == address(L2_ORACLE),
+            "ValidatorManager: Only L2OutputOracle can call this function"
+        );
+        _;
+    }
+
+    /**
+     * @notice Semantic version.
+     * @custom:semver 1.0.0
+     */
+    string public constant version = "1.0.0";
+
+    /**
+     * @notice Constructs the ValidatorManager contract.
+     *
+     * @param _constructorParams              Constructor parameters for AssetManager.
+     * @param _trustedValidator               Address of the trusted validator.
+     * @param _commissionRateMinChangeSeconds The minimum duration to change the commission rate in
+     *                                        seconds.
+     * @param _roundDurationSeconds           The duration of one submission round in seconds.
+     * @param _jailPeriodSeconds              The minimum duration to get out of jail in seconds.
+     * @param _jailThreshold                  The maximum allowed number of output non-submissions
+     *                                        before jailed.
+     */
+    constructor(
+        ConstructorParams memory _constructorParams,
+        address _trustedValidator,
+        uint128 _commissionRateMinChangeSeconds,
+        uint128 _roundDurationSeconds,
+        uint128 _jailPeriodSeconds,
+        uint128 _jailThreshold
+    ) AssetManager(_constructorParams) {
+        TRUSTED_VALIDATOR = _trustedValidator;
+        COMMISSION_RATE_MIN_CHANGE_SECONDS = _commissionRateMinChangeSeconds;
+
+        // Note that this value MUST be (SUBMISSION_INTERVAL * L2_BLOCK_TIME) / 2.
+        ROUND_DURATION_SECONDS = _roundDurationSeconds;
```

Comment:
Since the condition requires L2OutputOracle's variables, so it cannot be checked in constructor. If we move it to initialize function, the ROUND_DURATION_SECOND should be state variable, which increases gas cost during output submission. So how about check the condition in deploy script instead?

Reply:
I understand, that would be better